lib/engine/game/g_18_ireland/game.rb

```diff
@@ -188,6 +188,45 @@ def setup
           @corporations = corporations
         end
 
+        def get_par_prices(entity, _corp)
+          @game
+            .stock_market
+            .par_prices
+            .select { |p| p.price * 2 <= entity.cash }
+        end
+
+        def after_buy_company(player, company, price)
+          abilities(company, :shares) do |ability|
+            ability.shares.each do |share|
+              if share.president
+                # DKR is pared at the highest par price below
+                corporation = share.corporation
+                par_price = price / 2
+                share_price = @stock_market.par_prices.find { |sp| sp.price <= par_price }
+
+                @stock_market.set_par(corporation, share_price)
+                @share_pool.buy_shares(player, share, exchange: :free)
+                # Receives the bid money
+                @bank.spend(price, corporation)
+                after_par(corporation)
+                # And buys a 2 train
+                train = @depot.upcoming.first
+                buy_train(corporation, train, train.price)
```

Comment:
Its been a while since I last played 18Ireland. Does the corporation pay for the train or get it for free?

Reply:
Pays for it.